### PR TITLE
Mcu boot support

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
WIP

### CODE REVIEW

File: examples/lighting-app/nrfconnect/main/AppTask.cpp [hunk 1]
```diff
@@ -293,6 +323,14 @@ void AppTask::FunctionTimerEventHandler(AppEvent * aEvent)
     }
 }
 
+int AppTask::SoftwareUpdateConfirmationHandler(uint32_t offset, uint32_t size, void * arg)
+{
+    // For now just print update progress and confirm data chunk without any additional checks.
+    LOG_INF("Software update progress %d B / %d B", offset, size);
```
Comment (Damian-Nordic) [inline examples/lighting-app/nrfconnect/main/AppTask.cpp:329]: Use `PRIu32`?

**Review by Damian-Nordic — APPROVED** [review]
Nice work though would be good to have Greg take a look at the doc part before merging it. A few nits found.

File: examples/lock-app/nrfconnect/README.md [hunk 2]
```diff
@@ -83,6 +90,26 @@ sending the Thread network credentials from the CHIP controller to the CHIP
 device. As a result, device is able to join the Thread network and communicate
 with other Thread devices in the network.
 
+### Device Firmware Upgrade
```
Comment (doublemis1) [inline examples/lock-app/nrfconnect/README.md:93]: If you follow my comments in lighting-app README, then align it also in lock-app README

File: examples/lighting-app/nrfconnect/README.md [hunk 3]
```diff
@@ -318,6 +343,27 @@ kit you own:
...
     $ west build -b build-target -- -DOVERLAY_CONFIG=rpc.overlay
 
+### Building with Device Firmware Upgrade support
+
+To build the example with configuration enabling DFU, run the following command with _build-target_ replaced with the build target name of the Nordic Semiconductor's kit you own, for
+example `nrf52840dk_nrf52840`:
+
+> **_WARNING:_** Please do remember about replacing _build-target_ also in the PM_STATIC_YML_FILE path.
+
+    $ west build -b build-target -- -DOVERLAY_CONFIG=third_party/connectedhomeip/config/nrfconnect/app/overlay-dfu_support.conf -DPM_STATIC_YML_FILE="configuration/build-target/pm_static.yml"
```
[the remark this answers is not in the document]
Comment (kkasperczyk-no) [reply examples/lighting-app/nrfconnect/README.md:353]: I think it is not wrong, as this path is relative to the example directory, as you need to be there to call west build command

[the remark this answers is not in the document]
Comment (doublemis1) [reply examples/lighting-app/nrfconnect/README.md:353]: Ok, you are right, I have verified that Third_party / connectedhomeip is pointing directly to the root of the CHIP repository

**Pull request closed without merging**